cmake/pika_algorithms_generate_package.cmake

```diff
@@ -0,0 +1,98 @@
+# Copyright (c) 2018 Christopher Hinz
+# Copyright (c) 2014 Thomas Heller
+#
+# SPDX-License-Identifier: BSL-1.0
+# Distributed under the Boost Software License, Version 1.0. (See accompanying
+# file LICENSE_1_0.txt or copy at http://www.boost.org/LICENSE_1_0.txt)
+
+include(CMakePackageConfigHelpers)
+include(pika_algorithms_generate_package_utils)
+
+set(CMAKE_DIR
+    "cmake-${CMAKE_MAJOR_VERSION}.${CMAKE_MINOR_VERSION}"
+    CACHE STRING "directory (in share), where to put Findpika cmake module"
+)
+
+write_basic_package_version_file(
+  "${CMAKE_CURRENT_BINARY_DIR}/lib/cmake/pika-algorithms/pika-algorithms-config-version.cmake"
+  VERSION ${PIKA_ALGORITHMS_VERSION}
+  COMPATIBILITY AnyNewerVersion
+)
+
+# Export pika_algorithms_internal_targets in the build directory
+export(
+  TARGETS ${PIKA_ALGORITHMS_EXPORT_INTERNAL_TARGETS}
+  NAMESPACE pika_algorithms_internal::
+  FILE "${CMAKE_CURRENT_BINARY_DIR}/lib/cmake/pika-algorithms/pika_algorithms_internal_targets.cmake"
+)
+
+# Export pika_algorithms_internal_targets in the install directory
+install(
+  EXPORT pika_algorithms_internal_targets
+  NAMESPACE pika_algorithms_internal::
+  FILE pika_algorithms_internal_targets.cmake
+  DESTINATION ${CMAKE_INSTALL_LIBDIR}/cmake/pika-algorithms
+)
+
+# Export pika_algorithms_targets in the build directory
+export(
+  TARGETS ${PIKA_ALGORITHMS_EXPORT_TARGETS}
+  NAMESPACE pika::
```

Comment:
It is intentional, though I'm not entirely sure yet about this. I think it's nicer to have the same namespace as core `pika`. However, it does go against having `<package name>::<target name>` targets as a general rule. I'm slightly in favour of breaking the rule here (which is why I chose `pika::`), but happy to be convinced otherwise. What do you think?

Reply:
I've updated it to `pika-algorithms` now. The main target is then called `pika-algorithms::pika_algorithms`... still a somewhat weird mix, but probably right with all target names themselves being `snake_case` and the project/package name being `kebab-case`.